src/java/org/apache/cassandra/sensors/RequestSensorsFactory.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sensors;
+
+import org.apache.cassandra.config.CassandraRelevantProperties;
+import org.apache.cassandra.utils.FBUtilities;
+
+import static org.apache.cassandra.config.CassandraRelevantProperties.REQUEST_SENSORS_FACTORY;
+
+/**
+ * Provides a customizable factory to create a {@link RequestSensors} to track sensors per user request.
+ * Configured via the {@link REQUEST_SENSORS_FACTORY} system property, if unset a {@link NoOpRequestSensors} will be used.
+ * To activate tracking of request sensors, use {@link ActiveRequestSensors}.
+ */
+public interface RequestSensorsFactory
+{
+    RequestSensorsFactory instance = REQUEST_SENSORS_FACTORY.getString() == null ?
```

Comment:
I am not sure why is this needed?

Reply:
I was confused because I've seen test create `ActiveRequestSensorsFactory` directly. We should also mark that constructor as only visible for testing.